java/Junit4Example.java

```diff
@@ -0,0 +1,110 @@
+package tests.base;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.TestName;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.MutableCapabilities;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class BaseSauceJunit4  {
+    protected WebDriver driver;
+    public Boolean result;
+    
+     /**
+         * @Rule is a JUnit 4 annotation that defines specific test method behaviors.
+            For more information visit the wiki: https://github.com/junit-team/junit4/wiki/rules
+    */
+
+    @Rule
+    public TestName testName = new TestName() {
+        public String getMethodName() {
+            return String.format("%s", super.getMethodName());
+        }
+    };
+    
+     /**
+         * @Before is a JUnit 4 annotation that defines specific prerequisite test method behaviors.
+            In the example below we:
+            - Define Environment Variables for Sauce Credentials ("SAUCE_USERNAME" and "SAUCE_ACCESS_KEY")
+            - Define Chrome Options such as W3C protocol
+            - Define the "sauce:options" capabilities, indicated by the "sauceOpts" MutableCapability object
+            - Define the WebDriver capabilities, indicated by the "caps" DesiredCapabilities object
+            - Define the service URL for communicating with SauceLabs.com indicated by "sauceURL" string
+            - Set the URL to sauceURl
+            - Set the driver instance to a RemoteWebDriver
+            - Pass "url" and "caps" as parameters of the RemoteWebDriver
+            For more information visit the docs: https://junit.org/junit4/javadoc/4.12/org/junit/Before.html
+    */
+
+    @Before
+    public void setup() throws MalformedURLException {
+        String username = System.getenv("SAUCE_USERNAME");
+        String accessKey = System.getenv("SAUCE_ACCESS_KEY");
+        String methodName = testName.getMethodName();
+
+        ChromeOptions chromeOpts = new ChromeOptions();
+        chromeOpts.setExperimentalOption("w3c", true);
+
+        MutableCapabilities sauceOpts = new MutableCapabilities();
```

Comment:
Do you think it would be valuable to tie the key/value pairs here to line 64 below with a comment, so that users explicitly and immediately see that the stuff defined here is included as a capability later?

Reply:
I just added comments regarding `MutableCapabilities` and the `ChromeOptions` class extension.